ush/wafs/evs_wafs_atmos_stats_preparedata.sh

```diff
@@ -1,105 +1,181 @@
-	sourcedir=$COMINgfs/gfs.$VDATE/$cc/atmos
+    ccdir=$(( 10#$cc / 6 * 6 ))
+    ccdir="$(printf "%02d" $(( 10#$ccdir )) )"
+    sourcedir=$COMINgfs/gfs.$VDATE/$ccdir/atmos
+    sourcefile=$sourcedir/gfs.t${cc}z.gcip.f00.grib2
+
+    targetdir=$GRID_STAT_INPUT_BASE
+    targetfile=$GRID_STAT_INPUT_BASE/gfs.t${cc}z.gcip.f00.grib2
 
-	targetdir=$GRID_STAT_INPUT_BASE/gfs.$VDATE/$cc/atmos
-	sourcefile=$sourcedir/gfs.t${cc}z.gcip.f00.grib2
+    if [[ ! -f $targetfile ]] ; then
 	if [[ -f $sourcefile ]] ; then
-	    mkdir -p $targetdir
-	    ln -sf $sourcefile $targetdir/.
-	else
-	    if [ $SENDMAIL = YES ] ; then
-		export subject="GCIP Analysis Data Missing for EVS ${COMPONENT}"
-		echo "Warning: No GCIP analysis was available for valid date ${VDATE}${cc}" > mailmsg
-		echo “Missing file is $sourcefile” >> mailmsg
-		echo "Job ID: $jobid" >> mailmsg
-		cat mailmsg | mail -s "$subject" $MAILTO
+	    # For WAFS after implementation 2023-2024
+	    $WGRIB2 $sourcefile -match ":(812|696.8|595.2|506|392.7) mb" \
+		    -match "(parm=37:|ICESEV)" \
+		    -if ":812 mb" -set_lev "800 mb" -$end_wgrib2 \
+		    -if ":696.8 mb" -set_lev "700 mb" -$end_wgrib2 \
+		    -if ":595.2 mb" -set_lev "600 mb" -$end_wgrib2 \
+		    -if ":506 mb" -set_lev "500 mb" -$end_wgrib2 \
+		    -if ":392.7 mb" -set_lev "400 mb" -$end_wgrib2 \
+		    -grib $targetfile
+	    # For WAFS before implementation 2023-2024
+	    nrecords=`$WGRIB2 $targetfile | wc -l`
+	    if [[ $nrecords -le 4 ]] ; then
+		$WGRIB2 $sourcefile -match ":(800|700|600|500|400) mb" \
+			-match "(parm=37:|ICESEV)" \
+			-grib $targetfile
 	    fi
-	fi
-	
-	cc2=$(( 10#$cc + 3 ))
-	cc2="$(printf "%02d" $(( 10#$cc2 )) )"
-	targetdir=$GRID_STAT_INPUT_BASE/gfs.$VDATE/$cc2/atmos
-	sourcefile=$sourcedir/gfs.t${cc2}z.gcip.f00.grib2
-	if [[ -f $sourcefile ]] ; then
-            mkdir -p $targetdir
-            ln -sf $sourcefile $targetdir/.
 	else
 	    if [ $SENDMAIL = YES ] ; then
-		export subject="GCIP Analysis Data Missing for EVS ${COMPONENT}"
-		echo "Warning: No GCIP analysis was available for valid date ${VDATE}${cc2}" > mailmsg
-		echo “Missing file is $sourcefile” >> mailmsg
-		echo "Job ID: $jobid" >> mailmsg
-		cat mailmsg | mail -s "$subject" $MAILTO
+		export subject="$OBSERVATION Analysis Data Missing for EVS ${COMPONENT}"
+		echo "Warning: $OBSERVATION analysis is missing for $CENTER valid date ${VDATE}${cc}" > mailmsg.$OBSERVATION.$CENTER.$RESOLUTION
+		echo "Missing file is $sourcefile" >> mailmsg.$OBSERVATION.$CENTER.$RESOLUTION
+		echo "Job ID: $jobid" >> mailmsg.$OBSERVATION.$CENTER.$RESOLUTION
+		cat mailmsg.$OBSERVATION.$CENTER.$RESOLUTION | mail -s "$subject" $MAILTO
 	    fi
-        fi
-    done
+	    exit 99
+	fi
+    fi
+
 elif [[ $OBSERVATION = "GFS" ]] ; then
-    for cc in $inithours ; do
-        sourcedir=$COMINgfs/gfs.$VDATE/$cc/atmos
+    sourcedir=$COMINgfs/gfs.$VDATE/$cc/atmos
+    if [[ $icao2023 = yes ]] ; then
+	sourcefile=$sourcedir/gfs.t${cc}z.wafs.0p25.anl
```

Comment:
Hi, @YaliMao-NOAA ,

We're trying to figure out what icao2023 is supposed to represent.  Because in the prod case, we see that icao2023 is sometimes yes, sometimes no, but in the para case, we see that icao2023 is always yes.  What is this? 

In the dir: /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr365test/EVS/dev/drivers/scripts/stats/wafs

prod case:  jevs_wafs_stats.o94548738
para case:  jevs_wafs_stats.o94549935

@AliciaBentley-NOAA @ShelleyMelchior-NOAA

Reply:
Yes, currently icao2023=yes for 'para' case because of the new pressure levels. 

icao2023 should be the same value for u/v/t forecast and analysis for either case. After the incoming WAFS implementation, icao2023=yes because of the new pressure levels though it's going to be 'prod' case.

> in the para case, we see that icao2023 is always yes